modules/configuration/src/test/java/org/apache/ignite/internal/configuration/testframework/ConfigurationExtension.java

```diff
@@ -255,27 +292,79 @@ public long notificationCount() {
         return cfgRef.get();
     }
 
-    /**
-     * Looks for the annotated field inside the given test class.
-     *
-     * @return Annotated fields.
-     */
-    private static List<Field> getMatchingFields(Class<?> testClass) {
+    private static List<Field> getInjectConfigurationFields(Class<?> testClass) {
         return AnnotationSupport.findAnnotatedFields(
                 testClass,
                 InjectConfiguration.class,
-                field -> supportType(field.getType()),
+                field -> isNameEndsWithConfiguration(field.getType()),
                 HierarchyTraversalMode.TOP_DOWN
         );
     }
 
+    private static List<Field> getInjectRevisionListenerHolderFields(Class<?> testClass) {
+        return AnnotationSupport.findAnnotatedFields(
+                testClass,
+                InjectRevisionListenerHolder.class,
+                field -> isRevisionListenerHolder(field.getType()),
+                HierarchyTraversalMode.TOP_DOWN
+        );
+    }
+
+    private static boolean isNameEndsWithConfiguration(Class<?> type) {
+        return type.getCanonicalName().endsWith("Configuration");
+    }
+
+    private static boolean isRevisionListenerHolder(Class<?> type) {
+        return ConfigurationStorageRevisionListenerHolder.class.isAssignableFrom(type);
+    }
+
     /**
-     * Checks that instance of the given class can be injected by the extension.
-     *
-     * @param type Field or parameter type.
-     * @return {@code true} if value of the given class can be injected.
+     * Implementation for {@link ConfigurationExtension}.
      */
-    private static boolean supportType(Class<?> type) {
```

Comment:
The old name (`supportType()`) was good in the sense that it was telling *what was the meaning* of the postfix check. The new name is accurate, but it does not tell you *why* it checks for the postfix.

I would suggest to go back to the old approach (but change the name as this class now injects not only configurations, so my suggestion is to rename it to `supportsAsConfigurationType()`).

Reply:
Name you suggested makes sense.